src/com/twitter/intellij/pants/execution/PantsClasspathRunConfigurationExtension.java

```diff
@@ -63,13 +73,41 @@
      */
     params.setUseDynamicClasspath(true);
 
+    // Filter out all jars imported to the project with the exception of
+    // jars on the JDK path, IntelliJ installation paths, and plugin installation path,
+    // because Pants is already exporting all the runtime classpath in manifest.jar
+    String homePath = PathManager.getHomePath();
+    String jdkPath = params.getJdkPath();
+    String pluginPath = PathManager.getPluginsPath();
+
+    Set<String> pathsAllowed = Sets.newHashSet(jdkPath, homePath, pluginPath);
+    if (ApplicationManager.getApplication().isUnitTestMode()) {
+      pathsAllowed.add(
+        // /Users/yic/.ivy2/pants/com.intellij.sdk.community/idea_rt/jars/idea_rt-latest.jar ->
+        // /Users/yic/.ivy2/pants/com.intellij.sdk.community/
+        Optional.ofNullable(PluginManager.getPlugin(PluginId.getId("com.intellij")))
+          .map(s -> s.getPath().getParentFile().getParentFile().getParentFile().getAbsolutePath()).get()
+      );
+
+      pathsAllowed.add(
+        // At this point we know junit plugin is already enabled.
+        // //Users/yic/.ivy2/pants/com.intellij.junit-plugin/junit-rt/jars/junit-rt-latest.jar ->
+        // /Users/yic/.ivy2/pants/com.intellij.junit-plugin/
+        Optional.ofNullable(PluginManager.getPlugin(PluginId.getId("JUnit")))
+          .map(s -> s.getPath().getParentFile().getParentFile().getParentFile().getAbsolutePath()).get()
+      );
+    }
```

Comment:
I think this would be easier to follow if the code that builds `pathsAllowed` were broken out into a separate method.